apps/ai-image-tagging/frontend/src/components/AITagView/AITagView.js

```diff
@@ -28,41 +28,64 @@ export class AITagView extends React.Component {
       tags: props.entries.imageTags.getValue() || [],
       overwrite: true,
       isMissingImage: !props.entries.image.getValue(),
+      unsupportedImageType: false,
       isFetchingTags: false
     }
+
+    this.validateImage();
   }
 
   componentDidMount() {
     this.props.entries.image.onValueChanged(() => {
       this.setState(() => ({
         isMissingImage: !this.props.entries.image.getValue()
-      }))
+      }));
+      // always validate the image
+      this.validateImage();
     });
   }
 
+  validateImage = async () => {
+    const imageId = get(this.props.entries.image.getValue(), 'sys.id');
+    if (!imageId) { return true; }
+
+    const file = await this.props.space.getAsset(imageId);
+    const locale = this.props.locale;
+    const fullURL = get(file, `fields.file.${locale}.url`);
+    // test if file extension is PNG/JPEG/JPG
+    const isImageValid = new RegExp(/\.png|jpeg|jpg$/, 'i').test(fullURL);
```

Comment:
There is `contentType` property next to `url` which is more reliable (it's inferred from the binary rather than from the file extension).